packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1135,7 +1136,18 @@ export function getClientNamespace(
   entity: Namespace | Interface | Model | Enum | Union,
 ): string {
   const override = getScopedDecoratorData(context, clientNamespaceKey, entity);
-  if (override) return override;
+  if (override) {
```

Comment:
It would be good to add a comment here (or on the method doc) to document the intent with an example.